### DISCUSSION TITLE
Fetching XML file from `./public` (NextJS app)

### DISCUSSION THREAD

**Discussion by nifanic** [vercel/vercel#8010] [Help] [answered]
Hi folks,

I'm stuck with fetching static XML file (located in `./public`) after the app being deployed with Vercel. Image files (PNG) are served as expected.

Localhost serves public files without any issues.

Maybe I've missed something in the docs, and this behaviour is intentional. Please help.

---

```json
"dependencies": {
    "next": "12.1.6",
    "react": "18.1.0",
    "react-dom": "18.1.0",
}
```

1. Preview URL (desired structure) https://mcom-next-erv1q8bwt-masterpage.vercel.app/portfolio/ui/nomura-chartviewer/nomura_chartviewer-ui-en_us-2003v01.xml gives 404.
2. Preview URL (file moved to the root of `public` for test) https://mcom-next-erv1q8bwt-masterpage.vercel.app/nomura_chartviewer-ui-en_us-2003v01.xml gives 404.

<img width="783" alt="image" src="https://user-images.githubusercontent.com/28378773/175449746-1a14899e-07ec-40a9-abb5-718482224a9a.png">

My `./public` folder structure:

<img width="533" alt="image" src="https://user-images.githubusercontent.com/28378773/175448966-dd4a0a12-cddf-4c8b-bf85-987fbcca85d2.png">

**Comment by amyegan** [comment, marked as answer]
Hi @nifanic. It looks like there are a couple of problems going on here. The preview URL links are going to your 404 page because of a typo and "US" should be capitalized based on your file structure screenshots (happens to us all from time to time 😅). When I checked the updated URLs I can see the xml file, but it has a parsing error. It should start working correctly for you after you make those two changes.

Updated URL example: https://mcom-next-erv1q8bwt-masterpage.vercel.app/portfolio/ui/nomura-chartviewer/nomura_chartviewer-ui-en_US-2003v01.xml

**Reply by nifanic** [reply]
@amyegan, insane! 😆 I thank you to spot this typo!

I've thought all filenames are normalized (to be lower cased).

**Rhetorical question:** How come, my dev environment (I'm on Mac) served these XML files names formatted as lower cased? Perhaps, Linux (Vercel)…
